strings/palindrome.py

```diff
@@ -58,9 +79,29 @@ def is_palindrome_slice(s: str) -> bool:
     return s == s[::-1]
 
 
+def benchmark_function(name: str) -> None:
+    setup = f"from __main__ import test_data, {name}"
+    number = 100000
+    res = timeit(
+        f"all({name}(key) is value for key, value in test_data.items())",
+        setup=setup,
+        number=number,
+    )
+    print(f"{name:<35} finished {number} runs in {res:.5f} seconds")
+
+
 if __name__ == "__main__":
     for key, value in test_data.items():
         assert is_palindrome(key) is is_palindrome_recursive(key)
         assert is_palindrome(key) is is_palindrome_slice(key)
         print(f"{key:21} {value}")
     print("a man a plan a canal panama")
+
+    benchmark_function("is_palindrome")  # finished 100000 runs in 0.33785 seconds
+    benchmark_function(
+        "is_palindrome_traversal"
+    )  # finished 100000 runs in 0.70002 seconds
+    benchmark_function(
+        "is_palindrome_recursive"
+    )  # finished 100000 runs in 0.48514 seconds
+    benchmark_function("is_palindrome_slice")  # finished 100000 runs in 0.18703 seconds
```

Comment:
Easier to read... Sorted fastest to slowest.
```suggestion
    # finished 100000 runs in 0.18703 seconds
    benchmark_function("is_palindrome_slice")
    # finished 100000 runs in 0.33785 seconds
    benchmark_function("is_palindrome")
    # finished 100000 runs in 0.48514 seconds
    benchmark_function("is_palindrome_recursive")
    # finished 100000 runs in 0.70002 seconds
    benchmark_function("is_palindrome_traversal")
```